### ISSUE TITLE
Onion Peel on transformed rectilinear grid crashes parallel engine

### ISSUE THREAD

**Issue by aowen87** [aowen87/BAR#213]
Open chgcar_test_data/CHGCAR
Add PC plot
Apply OnionPeel operator 
Draw

parallel engine will crash.

Seems in part due to OnionPeel's use of vtkDataSetRemoveGhostCells.
commenting-out use of the filter allows OnionPeel to complete, but gives different
results than when run in serial.

Restructuring the code a bit:
old:
   ds = removeGhostCells->GetOutput();
   ds->Update();
new:
   removeGhostCells->Update();
   ds = removeGhostCells->GetOutput();

also removes the crash, but OnionPeel can fail with bad index.
eg. try seed cell of 97, which works in serial but in parallel
yields bad index error saying max is 49.

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. As such, not all 
information was able to be captured in the transition. Below is
a complete record of the original redmine ticket.

Ticket number: 632
Status: Resolved
Project: VisIt
Tracker: Bug
Priority: Normal
Subject: Onion Peel on transformed rectilinear grid crashes parallel engine
Assigned to: Kathleen Biagas
Category: 
Target version: 2.3
Author: Kathleen Biagas
Start: 02/24/2011
Due date: 
% Done: 0
Estimated time: 1.0
Created: 02/24/2011 12:42 pm
Updated: 02/24/2011 03:17 pm
Likelihood: 3 - Occasional
Severity: 4 - Crash / Wrong Results
Found in version: 2.2.0
Impact: 
Expected Use: 
OS: All
Support Group: Any
Description:
 Open chgcar_test_data/CHGCAR
Add PC plot
Apply OnionPeel operator 
Draw

parallel engine will crash.

Seems in part due to OnionPeel's use of vtkDataSetRemoveGhostCells.
commenting-out use of the filter allows OnionPeel to complete, but gives different
results than when run in serial.

Restructuring the code a bit:
old:
   ds = removeGhostCells->GetOutput();
   ds->Update();
new:
   removeGhostCells->Update();
   ds = removeGhostCells->GetOutput();

also removes the crash, but OnionPeel can fail with bad index.
eg. try seed cell of 97, which works in serial but in parallel
yields bad index error saying max is 49.

Comments:
Fixed the engine crash using code segment listed in bug ticket.The differing parallel/serial results are due to the automatic domain decomposition peformed by the format,and is exhibited by other format that do the same.Another ticket will be submitted for this.
